Documentation/building/versioning.md

```diff
@@ -0,0 +1,41 @@
+Package and File Versioning
+===========================
+
+We have now added package versioning and assembly file versioning when building the repo. This versioning works in both Windows and non-Windows platforms. Also, this versioning applies for both the managed and native libraries we produce. We had a few constrains to take into account, for example:
+- Version needed to be higher than the versions that we have already shipped.
+- There needs to be an ability to have multiple versions per day.
+- Versions need to be always increasing.
+- Version needs to be lower than [SHRT_MAX](https://www.google.com/?ion=1&espv=2#q=max%20short%20int) since the version is used as assembly file version which has that constrain.
+- (Nice to have) Version needs to be reproducible.
+- (Nice to have) We shouldn't have the need to check in a file containing the version or revision number.
+
+Taking into account all these constrains we came up with the new versioning. The version is calculated per build by [a targets file that lives in buildtools](https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets), and depending on what we are building(native/managed, windows/non-windows) it will generate versioning assets that will be used during the build to be able to produce build output with the right versions.
+
+How is the version calculated
+=============================
+
+The version is composed by two parts; the build number major and the build number minor. An example looks something like 00001.01, where the first five digits represent the build number major, and the two last digits after the dot represent the build number minor.
+
+Calculating BuildNumberMajor
+----------------------------
+
+The BuildNumberMajor is represented by 5 digits, and is determined by calculating the time that has happened between the latest commit's date and a SeedDate that gets passed in. The first portion of the BuildNumberMajor (first 3 digits) represent the number of month(s) since the SeedDate. The second portion of the BuildNumberMajor (last 2 digits) represent the day of the month of the latest commit's date. This part of the version is reproducible. 
```

Comment:
Isn't the SeedDate the actual commit date (https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets#L287) and you have a VersionComparisonDate (https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets#L284) which is the base date for doing the calculation? 

It is also worth calling out why we use 1996-04-01 as our comparison date (i.e. to ensure our version number is higher then what we have already shipped).